### ISSUE TITLE
SentryFlutter.Init will block the main thread in Profile/Release mode of version 8.11.0

### ISSUE THREAD

**Issue by muzipiao** [getsentry/sentry-dart#2499]
### Platform

Flutter Mobile Android

### Obfuscation

Disabled

### Debug Info

Disabled

### Doctor

```shell
lifei@lifeideMac-Studio ~ % flutter doctor -v
[✓] Flutter (Channel stable, 3.24.5, on macOS 15.1.1 24B91 darwin-arm64, locale
    zh-Hans-CN)
    • Flutter version 3.24.5 on channel stable at /Users/lifei/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision dec2ee5c1f (5 周前), 2024-11-13 11:13:06 -0800
    • Engine revision a18df97ca5
    • Dart version 3.5.4
    • DevTools version 2.37.3
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/lifei/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/lifei/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] VS Code (version 1.96.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (4 available)
    • PJX110 (mobile)                 • 8ccf7a2e              • android-arm64  •
      Android 14 (API 34)
    • macOS (desktop)                 • macos                 • darwin-arm64   •
      macOS 15.1.1 24B91 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         •
      macOS 15.1.1 24B91 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript •
      Google Chrome 131.0.6778.140

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Version

8.11.0

### Steps to Reproduce

SentryFlutter.Init will freeze the main thread in Release mode of version 8.11.0, and the appRunner method will not be called back. But if you lower the version of sentry_flutter to `sentry_flutter: 8.9.0`, it will run well.

SentryFlutter configuration:

```dart
class MainAppInit {
  static Future<void> buildSentryInit(String? languageType) async {
    final prefs = await SharedPreferences.getInstance();
    final int isAttachScreenshot = prefs.getInt(SentryConfigs.prefsAttachScreenshotKey) ?? 0;
    final int maxAttachmentSize = prefs.getInt(SentryConfigs.prefsMaxAttachmentSizeKey) ?? 5 * 1024 * 1024;
    await SentryFlutter.init(
      (options) {
        options.dsn = SentryConfigs.dsnURL;
        options.debug = false; 
        options.sendDefaultPii = true; 
        options.attachScreenshot = isAttachScreenshot == 1 ? true : false; 
        options.maxAttachmentSize = maxAttachmentSize; 
      },
      appRunner: () => runApp(SentryScreenshotWidget(child: MainApp(languageType: languageType))),
    );
    final String udidStr = await DeviceUtils.udidString();
    final String nameStr = await DeviceUtils.nameString();
    final SentryUser user = SentryUser(id: udidStr, name: nameStr);
    await Sentry.configureScope((scope) => scope.setUser(user));
  }
}
```

### Expected Result

The method `appRunner: () => runApp(SentryScreenshotWidget(child: MainApp(languageType: languageType)))` executes and initializes the home page

### Actual Result

Method `appRunner: () => runApp(SentryScreenshotWidget(child: MainApp(languageType: languageType)))` was not executed, unable to initialize home page

### Are you willing to submit a PR?

None

**Comment by buenaflor** [comment]
Hi we will release a fix for this on Monday

For a current workaround you can set

`SentryWidgetsFlutterBinding.ensureInitialized()` at the very beginning of your void main

**Comment by Tienisto** [comment]
This bug is very hidden and took me 2 hours to identify Sentry as the culprit.
The `appRunner` is actually called: You can check this with `print` statements.
However, the UI is not rerendered regardless of `setState`. The app "unfreezes" itself if you minimize and restore the app.

**Comment by stcojo** [comment]
Encountering this issue as well

**Comment by buenaflor** [comment]
@Tienisto apologies for that, release with the fix is on the way

**Comment by buenaflor** [comment]
I'll close this issue as a duplicate of https://github.com/getsentry/sentry-dart/issues/2491

**Comment by buenaflor** [comment]
Fix is now out with `8.11.1` please let us know if there are any more issues
